Issue #590 Automatic Site Backup

vladoleksiyenko opened the pull request:
# Description

Fixed the script that was previously mentioned in the closed PR 759. This along with the cron job template that will be mentioned in the #590 issue will allow automatic backup of the website.

Fixes #590

## Collaborators
- Matt Miss @MattMiss
- Vlad Oleksiyenko @vladoleksiyenko
- Brandon Viorato @brandonviorato
- Eugene Faison @Gene7Him

## Type of change

- [ ] Note merging this changes the database configuration.
- [x] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request]
- [x] I have removed text in ( ) from the issue request
- [x] You acknowledge that every person contributing to this work has signed the [OED Contributing License Agreement] and each author is listed in the Description section.

## Limitations

The cron template will be mentioned in the comments, that documentation will be need to be added to the webpage.

vladoleksiyenko committed:
added the backupScript with the fixes that Steve requested in the closed #792 PR

huss commented on src/scripts/backupScript.sh:
```diff
@@ -0,0 +1,22 @@
+#
+
+# Input the pathname for the desired backup directory
+# ie PATH="/home/<your_username>/path_to/database_dumps/dump.sql"
```
This isn't really an explicit path but a file location including the path. Maybe update the comment. Maybe also note that it will be the file listed but with the date and give an example.

Also, if the directory does not exist then this fails. At a minimum there should be a comment if that is the case. An alternative is to check if there it exists and create if it does not. Helpful for the first time this runs.

vladoleksiyenko replied:
Added an if statement that checks to see if the file path exists, and lets the user know if it exists. If not, it attempts to create the database_dumps directory. If this is unsuccessful, a message will then be displayed saying it failed. Thank you for bringing this up.

huss commented on src/scripts/backupScript.sh:
```diff
@@ -0,0 +1,22 @@
+# This could probably be programmatically populated. Currently needs to be set manually
+db_dump_path="/home/<your_username>/database_dumps" #INPUT REQUIRED
+
+# Generate a timestamp to append to the dump file. This line has errors at the moment.
```
This is working fine for me. Is there still an issue you see?

vladoleksiyenko replied:
Apologies, this was a comment that we forgot to remove originally, there aren't any issues with this. Thank you for noticing that.

huss commented on src/scripts/backupScript.sh:
```diff
@@ -0,0 +1,22 @@
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+# Input the pathname for the desired backup directory
```
It might be good to put in a comment at the start that the OED database Docker container must be running for this to work. It also assumes it is the only database container running (seems likely).

vladoleksiyenko replied:
I've included a message at the top that mentions this now, thank you for that suggestion.

huss reviewed (changes requested):
Thanks to @MattMiss, @vladoleksiyenko, @brandonviorato & @Gene7Him for addressing this issue. Testing found it mostly works as expected. I make a few comment to consider. ~~Also, OED is likely to create a cron job template to allow sites to automatically run this script at desired intervals. Let me know if you want to be involved in that and I can give more details.~~ (removed since I realized you had put a file on the issue)

vladoleksiyenko committed:
added a check for the backup directory, removed comment that mentions errors, indicated that OED docker database container must be running

huss committed:
add simple change

I forgot to mention this in my previous comments so I added a final output
statement to tell where the backup was placed.

huss reviewed (approved):
Thanks for addressing all the comments. Review and testing found it works as desired. Congratulations to  @MattMiss, @vladoleksiyenko, @brandonviorato & @Gene7Him on have this contribution accepted by OED.
